airflow/providers/core/sql/operators/sql.py

```diff
@@ -0,0 +1,315 @@
+            and "less_than" not in check_values
+            and "leq_to" not in check_values
+            and "equal_to" not in check_values
+        ):
+            raise ValueError(
+                "Please provide one or more of: less_than, leq_to, "
+                "greater_than, geq_to, or equal_to in the check's dict."
+            )
+
+        if "greater_than" in check_values and "less_than" in check_values:
+            if check_values["greater_than"] >= check_values["less_than"]:
+                raise ValueError(
+                    "greater_than should be strictly less than "
+                    "less_than. Use geq_to or leq_to for "
+                    "overlapping equality."
+                )
+
+        if "greater_than" in check_values and "leq_to" in check_values:
+            if check_values["greater_than"] >= check_values["leq_to"]:
+                raise ValueError(
+                    "greater_than must be strictly less than leq_to. "
+                    "Use geq_to with leq_to for overlapping equality."
+                )
+
+        if "geq_to" in check_values and "less_than" in check_values:
+            if check_values["geq_to"] >= check_values["less_than"]:
+                raise ValueError(
+                    "geq_to should be strictly less than less_than. "
+                    "Use leq_to with geq_to for overlapping equality."
+                )
+
+        if "geq_to" in check_values and "leq_to" in check_values:
+            if check_values["geq_to"] > check_values["leq_to"]:
+                raise ValueError("geq_to should be less than or equal to leq_to.")
+
+        if "greater_than" in check_values and "geq_to" in check_values:
+            raise ValueError("Only supply one of greater_than or geq_to.")
+
+        if "less_than" in check_values and "leq_to" in check_values:
+            raise ValueError("Only supply one of less_than or leq_to.")
+
+        if (
+            "greater_than" in check_values
+            or "geq_to" in check_values
+            or "less_than" in check_values
+            or "leq_to" in check_values
+        ) and "equal_to" in check_values:
+            raise ValueError(
+                "equal_to cannot be passed with a greater or less than "
+                "function. To specify 'greater than or equal to' or "
+                "'less than or equal to', use geq_to or leq_to."
+            )
+
+
+class SQLTableCheckOperator(BaseSQLOperator):
+    """
+    Performs one or more of the checks provided in the checks dictionary.
+    Checks should be written to return a boolean result.
+
+    :param table: the table to run checks on
+    :param checks: the dictionary of checks, e.g.:
+
+    .. code-block:: python
+
+        {
+            "row_count_check": {"check_statement": "COUNT(*) = 1000"},
+            "column_sum_check": {"check_statement": "col_a + col_b < col_c"},
+        }
+
+    :param conn_id: the connection ID used to connect to the database
```

Comment:
I think this should be indented further, not sure though, can you post a screenshot of how this looks in docs

Reply:
Is that built if I run `breeze build-docs`? I can't seem to find a preview anywhere (and not sure that putting in a separate markdown file to preview is going to be an accurate reflection of this).

Reply:
Yes. when you run `breeze build-docs --package-filter apache-airlfow-provider-core-sql` and run `./docs/start-docs-server.sh` or something - you should see the docs (it's described in contributing docs I am quite sure of that)